src/test/scala/SparkIntegrationTest.scala

```diff
@@ -180,6 +180,108 @@ class SparkIntegrationTest
+
+    articles.write
+      .format("io.weaviate.spark.Weaviate")
+      .option("scheme", "http")
+      .option("host", "localhost:8080")
+      .option("className", "Article")
+      .mode("append")
+      .save()
+
+    val results = client.data().objectsGetter()
+      .withClassName("Article")
+      .run()
+
+    if (results.hasErrors) {
+      println("Error getting Articles" + results.getError.getMessages)
+    }
+
+    assert(results.getResult.size == 1)
+    val props = results.getResult.get(0).getProperties
+    assert(props.get("title") == "Sam")
+    assert(props.get("content") == "Sam and Sam")
+    assert(props.get("wordCount") == 3)
+    assert(props.get("keywords") == List("yo", "hey").asJava)
+    WeaviateDocker.deleteClass()
+  }
+
+  test("Article with double array") {
+    WeaviateDocker.createClass(Property.builder()
+      .dataType(List[String]("number[]").asJava)
+      .name("doubles")
+      .build()
+    )
+    import spark.implicits._
+    val articles = Seq(ArticleWithDoubleArray("Sam", "Sam and Sam", 3, Array(1.0, 2.0, 3.0))).toDF
+
+    articles.write
+      .format("io.weaviate.spark.Weaviate")
+      .option("scheme", "http")
+      .option("host", "localhost:8080")
+      .option("className", "Article")
+      .mode("append")
+      .save()
+
+    val results = client.data().objectsGetter()
+      .withClassName("Article")
+      .run()
+
+    if (results.hasErrors) {
+      println("Error getting Articles" + results.getError.getMessages)
+    }
+
+    assert(results.getResult.size == 1)
+    val props = results.getResult.get(0).getProperties
+    assert(props.get("title") == "Sam")
+    assert(props.get("content") == "Sam and Sam")
+    assert(props.get("wordCount") == 3)
+    assert(props.get("doubles") == List(1.0, 2.0, 3.0).asJava)
+    WeaviateDocker.deleteClass()
+  }
+
+  test("Article with int array") {
+    WeaviateDocker.createClass(Property.builder()
+      .dataType(List[String]("int[]").asJava)
+      .name("ints")
+      .build()
+    )
+    import spark.implicits._
+    val articles = Seq(ArticleWithIntArray("Sam", "Sam and Sam", 3, Array(1, 2, 3))).toDF
+
+    articles.write
+      .format("io.weaviate.spark.Weaviate")
+      .option("scheme", "http")
+      .option("host", "localhost:8080")
+      .option("className", "Article")
+      .mode("append")
+      .save()
+
+    val results = client.data().objectsGetter()
+      .withClassName("Article")
+      .run()
+
+    if (results.hasErrors) {
+      println("Error getting Articles" + results.getError.getMessages)
+    }
+
+    assert(results.getResult.size == 1)
+    val props = results.getResult.get(0).getProperties
+    assert(props.get("title") == "Sam")
+    assert(props.get("content") == "Sam and Sam")
+    assert(props.get("wordCount") == 3)
+    assert(props.get("ints") == List(1.0, 2.0, 3.0).asJava) // TODO: why are these coming back as dubs?
```

Comment:
this seems whack but probably not the end of the world

Reply:
@antas-marcin might have some thoughts on why Weaviate is returning doubles when we try get an object with an array of ints. I suspect the java client might be the culprit and will check that. I remember a similar issue in the Java client before.

I'm good with merging as is for now with the TODO